src/main/scala/app/ControllerBase.scala

```diff
@@ -34,8 +35,35 @@ abstract class ControllerBase extends ScalatraFilter
     val context      = request.getServletContext.getContextPath
     val path         = httpRequest.getRequestURI.substring(context.length)
 
+    var account = httpRequest.getSession.getAttribute(Keys.Session.LoginAccount).asInstanceOf[Account]
+    if(account == null) {
```

Comment:
please avoid calling isDefined/get on Option, this can be done in a safer way using foreach or a for-comprehension